geode-core/src/main/java/org/apache/geode/internal/metrics/CacheLifecycleMetricsSession.java

```diff
@@ -68,7 +88,14 @@ public void cacheClosed(InternalCache cache) {
     cacheLifecycle.removeListener(this);
 
     for (MetricsPublishingService metricsPublishingService : metricsPublishingServices) {
-      metricsPublishingService.stop();
+      try {
+        metricsPublishingService.stop();
+      } catch (VirtualMachineError e) {
+        SystemFailure.initiateFailure(e);
+        throw e;
+      } catch (Error | RuntimeException e) {
+        logError(errorLogger, metricsPublishingService.getClass().getName(), e);
```

Comment:
This logs "Error invoking start," even though we invoked `stop()`. Perhaps pass the method name to `logError()`?

Reply:
Fixed